File: 27.Remove Element/L27_removeElement.py
class Solution:
    def remove_element(self,nums:List[int],val:int)->int:
        p1,p2 = 0,0 # double pointers
        while p2 < len(nums):
            #get the valid element to the front
            if nums[p2] != val:
                nums[p1],nums[p2] = nums[p2],nums[p1]
                p1 += 1 # save to update the front pointer
            p2 += 1
        return p1
    
    #O(n),O(1)

File: 27.Remove Element/L27_removeElement.java
public class L27_removeElement {
    public int removeElement(int[] nums, int val){
        int p1=0, p2 =0;
        int cur;
        while(p2 < nums.length){
            if(nums[p2] != val){
                cur = nums[p1];
                nums[p1] = nums[p2];
                nums[p2] = cur;
                p1 ++;
            }
            p2 ++;
        }
        return p1;
    }
}
